src/rgw/driver/rados/rgw_cfg_store.cc

```diff
@@ -0,0 +1,64 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab ft=cpp
+
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2019 Red Hat, Inc.
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation. See file COPYING.
+ *
+ */
+
+#include <errno.h>
+#include <stdlib.h>
+#include <dlfcn.h>
+#include <system_error>
+#include <unistd.h>
+#include <sstream>
+
+#include "common/errno.h"
+
+#include "rgw_sal.h"
+#include "rgw_sal_rados.h"
+#include "driver/rados/config/store.h"
+#include "driver/json_config/store.h"
+#include "rgw_d3n_datacache.h"
+
+#ifdef WITH_RADOSGW_DBSTORE
+#include "rgw_sal_dbstore.h"
+#include "driver/dbstore/config/store.h"
+#endif
+
+#define dout_subsys ceph_subsys_rgw
+
+auto DriverManager::create_config_store(const DoutPrefixProvider* dpp,
+                                        std::string_view type)
+  -> std::unique_ptr<rgw::sal::ConfigStore>
```

Comment:
looks like it's a mistake. DriverManager::create_config_store() is also in ../rgw/rgw_sal.cc. It will be removed in the next push

Reply:
okay, i think it would make sense for `create_config_store()` to be a separate (optional) sal entrypoint. that should also help with the dbstore case below